cache/class-vary-cache.php

```diff
@@ -414,24 +416,28 @@ private static function parse_nocache_cookie() {
 	 */
 	private static function parse_group_cookie() {
 		// If the cache layer supplies a decrypted segmentation header, use that instead of decrypting it again.
-		if ( self::is_encryption_enabled() && isset( $_SERVER[ self::HEADER_AUTH ] ) &&  ! empty( $_SERVER[ self::HEADER_AUTH ] ) ) {
+		if ( self::is_encryption_enabled() && ! empty( $_SERVER[ self::HEADER_AUTH ] ) ) {
+			// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
 			$cookie_value = $_SERVER[ self::HEADER_AUTH ];
-		} elseif ( self::is_encryption_enabled() && ! empty( $_COOKIE[ self::COOKIE_AUTH ] ) ) {
+		} elseif ( self::is_encryption_enabled() && ! empty( $_COOKIE[ self::COOKIE_AUTH ] ) && isset( $_SERVER['HTTP_COOKIE'] ) ) {
 			// If the header auth isn't set (in case of a logged-in user), fall back to decrypting the cookie itself.
 			$auth_cookie = null;
 			// $_COOKIE is automatically urldecoded, so we need to search through the $_SERVER version to get the unencoded one.
-			foreach( explode( '; ', $_SERVER[ 'HTTP_COOKIE' ] ) as $rawcookie ) {
+			// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			foreach ( explode( '; ', $_SERVER['HTTP_COOKIE'] ) as $rawcookie ) {
 				list( $k, $v ) = explode( '=', $rawcookie, 2 );
-				if( self::COOKIE_AUTH === $k ) {
+				if ( self::COOKIE_AUTH === $k ) {
 					$auth_cookie = $v;
 					break;
 				}
 			}
 
-			$value = ltrim( $auth_cookie, VIP_GO_APP_ID . '.' ); // remove the site prefix
+			// Remove the site prefix
+			$value        = ltrim( $auth_cookie, VIP_GO_APP_ID . '.' );
```

Comment:
Had to move the comment to the top because of the weird complaint:

> Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native "phpcs:ignore Standard.Category.SniffName.ErrorCode" annotations instead. Found: // Remove the site prefix\nPHPCS(WordPress.NamingConventions.PrefixAllGlobals.DeprecatedWhitelistCommentFound)

Reply:
As a side note: is this code correct? `ltrim()` will not remove the prefix; it will remove all leading characters specified by its second argument. Thus, if `VIP_GO_APP_ID` is `1234`, the code will remove `1234321.` from the prefix, which is *probably* not the intended behavior.

Reply:
> Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native "phpcs:ignore Standard.Category.SniffName.ErrorCode" annotations instead. Found: // Remove the site prefix\n PHPCS(WordPress.NamingConventions.PrefixAllGlobals.DeprecatedWhitelistCommentFound)

That weird complaint seems to be a false positive. However, that rule seems to be no longer there as of https://github.com/WordPress/WordPress-Coding-Standards/pull/1908/files, so in future versions of WPCS (3.0+), it won't be flagged.

Reply:
Re: the `ltrim()`, @dchymko can you speak to this? I noticed it was added in https://github.com/Automattic/vip-go-mu-plugins/pull/1285/files.